ChronoVisor/include/ClientInfo.h

```diff
@@ -5,14 +5,47 @@
 #ifndef CHRONOLOG_CLIENTINFO_H
 #define CHRONOLOG_CLIENTINFO_H
 
+#include <sstream>
+
+class Story;
 
 class ClientInfo {
 public:
     ClientInfo() = default;
 
+    friend std::ostream &operator<<(std::ostream &out, const ClientInfo &r) {
+        std::stringstream ss;
```

Comment:
I'd rather avoid using friend functions/members if at all possible. And it's possible in this case , as you have to_string() function and you also have all the members of ClientInfo defined as public. you can define operator<<  outside of ClientInfo class and it doesn't need to be a friend.